src/query/graphite/native/aggregation_functions.go

```diff
@@ -561,24 +561,54 @@ func applyByNode(ctx *common.Context, seriesList singlePathSpec, nodeNum int, te
 //
 //    sumSeries(foo.by-function.server1.*.cpu.load5),sumSeries(foo.by-function.server2.*.cpu.load5),...
 func groupByNode(ctx *common.Context, series singlePathSpec, node int, fname string) (ts.SeriesList, error) {
-	metaSeries := make(map[string][]*ts.Series)
-	for _, s := range series.Values {
-		parts := strings.Split(s.Name(), ".")
+	return groupByNodes(ctx, series, fname, []int{node}...)
+}
+
+func findFirstMetricExpression(seriesName string) string {
+	idxOfRightParen := strings.Index(seriesName, ")")
+	substring := seriesName[:idxOfRightParen]
+	idxOfLeftParen := strings.LastIndex(substring, "(")
+	return seriesName[idxOfLeftParen+1:idxOfRightParen]
+}
+
+func getParts(series *ts.Series) []string {
+	seriesName := series.Name()
+	if strings.Contains(seriesName, ")") {
+		seriesName = findFirstMetricExpression(seriesName)
+	}
+
+	return strings.Split(seriesName, ".")
 
-		n := node
+}
+func getAggregationKey(series *ts.Series, nodes []int) string {
+	parts := getParts(series)
+
+	var keys []string
+	for _, n := range nodes {
 		if n < 0 {
 			n = len(parts) + n
 		}
 
-		if n >= len(parts) || n < 0 {
-			return aggregate(ctx, series, fname)
+		if n < len(parts) {
+			keys = append(keys, parts[n])
 		}
+	}
+	return strings.Join(keys, ".")
+}
+
+func getMetaSeriesGrouping(seriesList singlePathSpec, nodes []int) map[string][]*ts.Series {
+	metaSeries := make(map[string][]*ts.Series)
 
-		key := parts[n]
-		metaSeries[key] = append(metaSeries[key], s)
+	if len(nodes) > 0 {
+		for _, s := range seriesList.Values {
+			key := getAggregationKey(s, nodes)
+			if key != "" {
```

Comment:
`graphite-web` does not seem to perform this behavior,  let's make this match `graphite-web` and always add the aggregation to the key.